xla/service/gpu/model/indexing_analysis.cc

```diff
@@ -684,7 +684,7 @@ std::vector<int64_t> ToTransposeDimensions(const Layout& l) {
   return out;
 }
 
-llvm::SmallVector<AffineExpr, 4> DelinearizeInBoundsIndex(
+llvm::SmallVector<AffineExpr, 6> DelinearizeInBoundsIndex(
     mlir::AffineExpr linear, absl::Span<const int64_t> sizes,
     absl::Span<const int64_t> strides) {
   llvm::SmallVector<AffineExpr> result;
```

Comment:
Alternatively, if 4 is actually the right number of element to inline (e.g. it's the max number of dimensions), update the declaration of `result` to match it:
```suggestion
  llvm::SmallVector<AffineExpr, 4> result;
```

Am I missing something? Does your interpretation match mine?

Reply:
sync'd offline, the actual number of this is 6 right now but it could grow in the future, so using a generic array type without a fixed range is more reasonable here.